<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xmlns="http://www.springframework.org/schema/beans"
       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">

    <bean
            class="org.springframework.beans.factory.config.PropertyPlaceholderConfigurer">
        <property name="location" value="classpath:conf/jdbc-test.properties"/>
    </bean>

    <import resource="classpath:applicationContext-datasource-test.xml"/>

    <bean
            class="org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean"
            id="entityManagerFactory">
        <property name="persistenceXmlLocation">
            <value>classpath:conf/persistence-test.xml</value>
        </property>
        <property name="dataSource" ref="dataSource"/>
    </bean>

    <bean class="org.springframework.orm.jpa.JpaTransactionManager"
          id="transactionManager">
        <property name="entityManagerFactory" ref="entityManagerFactory"/>
    </bean>

    <!-- convert database exception to Spring DataAccessException for every
        class with @Repository annotation -->
    <bean
            class="org.springframework.dao.annotation.PersistenceExceptionTranslationPostProcessor"/>

    <bean id="genericDao" class="net.waryss.mtools.bluepurse.dao.impl.GenericDao"/>

</beans>
